### PR TITLE
N2n regulation groups

### PR DESCRIPTION
**Pull request by LKajan**
Implements n2n relations between plan objects and regulation groups.
- Removes lifecycle status change propagation from plan objects to regulation groups
- Plan lifecycle status changes are propagated to regulation groups

This PR is based on #385
Depends on:
- #401 
- #402

All these three should be merged before this one

### CODE REVIEW

File: database/base.py [hunk 1]
```diff
@@ -186,5 +180,14 @@ def plan(cls) -> Mapped[VersionedBase]:  # noqa
 
     # class reference in abstract base class, with backreference to class name:
     @declared_attr
-    def plan_regulation_group(cls) -> Mapped[VersionedBase]:  # noqa
-        return relationship("PlanRegulationGroup", backref=f"{cls.__tablename__}s")
+    def plan_regulation_groups(cls) -> Mapped[List[VersionedBase]]:  # noqa
+        return relationship(
+            "PlanRegulationGroup",
+            secondary="hame.regualtion_group_association",
```
Comment (Rikuoja) [inline database/base.py:186]: Typo again!

File: database/models.py [hunk 2]
```diff
@@ -14,8 +15,82 @@
     unique_str,
 )
 from shapely.geometry import MultiLineString, MultiPoint, MultiPolygon
-from sqlalchemy import ForeignKey
+from sqlalchemy import Column, ForeignKey, Table, Uuid
 from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy.sql import func
+
+regualtion_group_association = Table(
```
Comment (Rikuoja) [inline database/models.py:22]: Typo

**Review by Rikuoja — DISMISSED** [review]
Looks really good, just a typo.

File: database/models.py (hunk 2 above)
Comment (LKajan) [reply database/models.py]: fixed

**Pull request merged**